src/Domain/LeanCode.DomainModels/Model/SId.cs

```diff
@@ -0,0 +1,162 @@
+using System.Reflection;
+using LeanCode.DomainModels.Serialization;
+
+namespace LeanCode.DomainModels.Model
+{
+    /// <summary>A stronly typed id consisting of a guid prefixed with a lower-cased name of an entity.</summary>
+    /// <remarks>Use <see cref="IdSlugAttribute"/> to override default entity prefix.</remarks>
+    [TypedIdConverter]
+    public readonly struct SId<TEntity> : IEquatable<SId<TEntity>>, IComparable<SId<TEntity>>
+        where TEntity : class, IIdentifiable<SId<TEntity>>
+    {
+        private const int GuidLength = 32;
+        private const char Separator = '_';
+        private const int MaxTypeLength = 10;
+        private static readonly string TypeName = ExtractType();
+
+        public static readonly int RawLength = GuidLength + 1 + TypeName.Length;
+        public static readonly SId<TEntity> Empty = new(Guid.Empty);
+
+        private readonly string? value;
+
+        public string Value => value ?? Empty.Value;
+        public bool IsEmpty => value is null || value == Empty;
+
+        private SId(string v)
+        {
+            value = v ?? throw new ArgumentNullException(nameof(v));
```

Comment:
By the way, do we need this check here? `v` is already marked as non–null and the constructor is `private` so we shouldn't have to verify twice.